app/connector/google-sheets/src/main/java/io/syndesis/connector/sheets/GoogleSheetsVerifierExtension.java

```diff
@@ -0,0 +1,63 @@
+package io.syndesis.connector.sheets;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.services.sheets.v4.Sheets;
+import org.apache.camel.CamelContext;
+import org.apache.camel.component.extension.verifier.ResultBuilder;
+import org.apache.camel.component.extension.verifier.ResultErrorBuilder;
+import org.apache.camel.component.google.sheets.BatchGoogleSheetsClientFactory;
+import org.apache.camel.component.google.sheets.GoogleSheetsClientFactory;
+import org.apache.camel.component.google.sheets.GoogleSheetsConfiguration;
+
+
+/**
+ * FIXME Remove this class when Camel dependency gets updated
+ **/
+public class GoogleSheetsVerifierExtension
+    extends org.apache.camel.component.google.sheets.GoogleSheetsVerifierExtension {
+    public GoogleSheetsVerifierExtension(String defaultScheme) {
+        super(defaultScheme);
+    }
+
+    public GoogleSheetsVerifierExtension(String defaultScheme, CamelContext context) {
+        super(defaultScheme, context);
+    }
+
+    @Override
+    protected Result verifyConnectivity(Map<String, Object> parameters) {
+        ResultBuilder builder = ResultBuilder.withStatusAndScope(Result.Status.OK, Scope.CONNECTIVITY);
+
+        try {
+            try {
+                GoogleSheetsConfiguration configuration =
+                    this.setProperties(new GoogleSheetsConfiguration(), parameters);
+                GoogleSheetsClientFactory clientFactory = new BatchGoogleSheetsClientFactory();
+                Sheets client = clientFactory.makeClient(configuration.getClientId(), configuration.getClientSecret(),
+                    configuration.getApplicationName(), configuration.getRefreshToken(),
+                    configuration.getAccessToken());
+                client.spreadsheets().get(Optional.ofNullable(parameters.get("spreadsheetId")).map(Object::toString).orElse(UUID.randomUUID().toString())).execute();
+            } catch (GoogleJsonResponseException e) {
+                Integer code = e.getStatusCode();
+                //We may have a "404 spreadsheet no found" which is fine
+                //What we don't want is a bad credential error or similar
+                if (code != 404) {
+                    //Move to outer error
+                    throw e;
+                }
+            }
+        } catch (Exception var6) {
+            var6.printStackTrace();
```

Comment:
`printStackTrace()` writes to `System.out` I would rather log via logging framework. In general I doubt that we need to log the stacktrace here (maybe some leftover from debugging session?).

Reply:
Right, leftover :)